ArduPlane/quadplane.cpp

```diff
@@ -701,45 +697,44 @@ bool QuadPlane::setup(void)
 
     const static char *strUnableToAllocate = "Unable to allocate";
     if (!motors) {
-        hal.console->printf("%s motors\n", strUnableToAllocate);
-        goto failed;
+        AP_BoardConfig::config_error("%s motors", strUnableToAllocate);
     }
 
     AP_Param::load_object_from_eeprom(motors, motors_var_info);
 
     // create the attitude view used by the VTOL code
     ahrs_view = ahrs.create_view(rotation, ahrs_trim_pitch);
     if (ahrs_view == nullptr) {
-        goto failed;
+        AP_BoardConfig::config_error("%s ahrs_view", strUnableToAllocate);
     }
 
     attitude_control = new AC_AttitudeControl_Multi(*ahrs_view, aparm, *motors, loop_delta_t);
     if (!attitude_control) {
-        hal.console->printf("%s attitude_control\n", strUnableToAllocate);
-        goto failed;
+        AP_BoardConfig::config_error("%s attitude_control", strUnableToAllocate);
     }
     AP_Param::load_object_from_eeprom(attitude_control, attitude_control->var_info);
     pos_control = new AC_PosControl(*ahrs_view, inertial_nav, *motors, *attitude_control);
     if (!pos_control) {
-        hal.console->printf("%s pos_control\n", strUnableToAllocate);
-        goto failed;
+        AP_BoardConfig::config_error("%s pos_control", strUnableToAllocate);
     }
     AP_Param::load_object_from_eeprom(pos_control, pos_control->var_info);
     wp_nav = new AC_WPNav(inertial_nav, *ahrs_view, *pos_control, *attitude_control);
     if (!wp_nav) {
-        hal.console->printf("%s wp_nav\n", strUnableToAllocate);
-        goto failed;
+        AP_BoardConfig::config_error("%s wp_nav", strUnableToAllocate);
     }
     AP_Param::load_object_from_eeprom(wp_nav, wp_nav->var_info);
 
     loiter_nav = new AC_Loiter(inertial_nav, *ahrs_view, *pos_control, *attitude_control);
     if (!loiter_nav) {
-        hal.console->printf("%s loiter_nav\n", strUnableToAllocate);
-        goto failed;
+        AP_BoardConfig::config_error("%s loiter_nav", strUnableToAllocate);
     }
     AP_Param::load_object_from_eeprom(loiter_nav, loiter_nav->var_info);
 
     motors->init((AP_Motors::motor_frame_class)frame_class.get(), (AP_Motors::motor_frame_type)frame_type.get());
+
+    if (!motors->initialised_ok()) {
+        AP_BoardConfig::config_error("unknown Q_FRAME_TYPE %d", frame_type.get());
```

Comment:
Should this be %u?  Similarly elsewhere.

Reply:
Is AP_Int8 actually unsigned? It looks to me like the type is int8_t.

Reply:
That's an artifact of the storage.  We use them as unsigned.  I'm not that fussed here, but if we did get a Q_FRAME_TYPE of 137 it would appear here as a negative number which would be rather odd.

Reply:
now passing frame_class and frame_type as unsigned